app/code/Magento/CatalogUrlRewrite/Observer/AfterImportDataObserver.php

```diff
@@ -487,7 +487,7 @@ private function currentUrlRewritesRegenerate()
             $url = $currentUrlRewrite->getIsAutogenerated()
                 ? $this->generateForAutogenerated($currentUrlRewrite, $category)
                 : $this->generateForCustom($currentUrlRewrite, $category);
-            $urlRewrites = $url + $urlRewrites;
+            $urlRewrites = array_merge($urlRewrites, $url);
```

Comment:
It's strongly discouraged to use `array_merge` in a loop due to performance issues. The right solution would be:
```php
        $urlRewritesRaw = [[]];
        foreach ($currentUrlRewrites as $currentUrlRewrite) {
            $category = $this->retrieveCategoryFromMetadata($currentUrlRewrite);
            if ($category === false) {
                continue;
            }
            $url = $currentUrlRewrite->getIsAutogenerated()
                ? $this->generateForAutogenerated($currentUrlRewrite, $category)
                : $this->generateForCustom($currentUrlRewrite, $category);
            $urlRewritesRaw[] = $url;
        }

        $urlRewrites = array_merge(...$urlRewritesRaw);
```

That's written from my head, however, should work fine.

Reply:
You may be right with performance however, every time array keys are equal for '$url' array in the loop. Because of this only the last "$url" array is returned with the current coding. This has to be a bug for multi store websites.

Agree with your code. Just adding, `$urlRewritesRaw[] = $url;` 

Should be fine instead of `$urlRewrites = array_merge($urlRewrites, $url);`